### ISSUE TITLE
swagger-maven-plugin resolves wrong OpenAPI spec within maven multi module project

### ISSUE THREAD

**Issue by mtamme** [swagger-api/swagger-core#3019]
Hi!

I have a maven multi module project with two maven module containing JAX-RS resources. My swagger-maven-plugin config for each maven module looks as follows:

```
<plugin>
    <groupId>io.swagger.core.v3</groupId>
    <artifactId>swagger-maven-plugin</artifactId>
    <version>2.0.5</version>
    <executions>
        <execution>
            <phase>compile</phase>
            <goals>
                <goal>resolve</goal>
            </goals>
            <configuration>
                <openapiFilePath>${project.basedir}/src/main/swagger/openapi.json</openapiFilePath>
                <outputFileName>openapi</outputFileName>
                <outputPath>${project.build.directory}/generated-resources/swagger</outputPath>
                <outputFormat>JSON</outputFormat>
                <prettyPrint>true</prettyPrint>
                <resourcePackages>
                    <resourcePackage>[java-resource-root-package-name]</resourcePackage>
                </resourcePackages>
            </configuration>
        </execution>
    </executions>
</plugin>
```
When I do a `mvn clean install` on each maven module everything is fine and two different openapi.json files are resolved by the maven plugin. When I perform the same build on the whole maven project the Swagger maven plugin resolves the JAX-RS resources from the first module (regarding the build order) correctly. For the second JAX-RS resource module I actually get the same openapi.json file resolved as for the first JAX-RS resource module.
Looks like the Swagger maven plugin doesn't scan for the JAX-RS resources in the second module and just resolves the second openapi.json spec based on the JAX-RS resources from the first module.

I saw that you manipulate the classpath within the Swagger maven plugin. Looks like this interferes with the maven class loader scopes/handling.

I am using the following versions:
Oracle Java 1.8.0 Update 192
Apache Maven 3.6.0
maven-compiler-plugin 3.8.0
swagger-maven-plugin 2.0.5

Thanks,
Martin

**Comment by frantuma** [comment]
This seems to be caused by same issue of #3013, which is fixed in latest `2.0.6-SNAPSHOT` available on sonatype, please give it a go and close ticket if it solves your issue

**Comment by mtamme** [comment]
I just cloned the master branch (51bfc35facc819b26a9d8604be6354c1dbf60c15, Version 2.0.6-SNAPSHOT) and built the whole swagger-core project and it worked as expected.
